tendermint/src/lite/verifier.rs

```diff
@@ -127,7 +127,8 @@ where
         };
 
         // check vote is valid from validator
-        if !val.verify_signature(vote.sign_bytes(), vote.signature()) {
+        let sign_bytes = vote.sign_bytes();
+        if !val.verify_signature(&sign_bytes, vote.signature()) {
```

Comment:
Lines 121-133 are duplicated in verify_commit_trusting() and here. Maybe a function that takes a validator and a vote and returns error be useful to both? I know in verify_commit_trusting() we do an extra check but that could be done before or after.

Reply:
Good catch. There is a subtle difference in those lines though: in one case the validator is looked up in the set in the other it part of the looped elements (also some of the lines are part of the control flow via `None => continue,`).  This could be deduplicated (completely) if we do not `zip` the iterators and do a lookup in both cases. I'll leave this as is for now.

Reply:
> There is a subtle difference in those lines though: in one case the validator is looked up in the set in the other it part of the looped elements (also some of the lines are part of the control flow via None => continue,). 

Yes, I saw that but was thinking that once there is a validator and a vote the checks should be the same (most the loop body). But I looked closer and it's not much left in common to justify a separate function.

Another question, for `verify_commit_full()` we do assume: `The vals and commit have a 1-to-1 correspondence`; is this checked elsewhere? I was looking for a check that verifies that `val.address == vote.validator_address`. Or `val.verify_signature()` will take care of it?

Reply:
Also, for `vals_iter.zip(commit_iter)`, I believe this works as we want as long as there is no nil validator in the set (I think this is the case but I have a hard time navigating the Rust code :(  ) ...otherwise according to the doc: ` If the first iterator returns None, zip will short-circuit and next will not be called on the second iterator.`

Reply:
The only check is that length matches. Also AFAIR the commits / validators are sorted. That means if the the len matches there is the desired correspondence. But I'll double check and follow up in #63. It seems this is more confusing than helpful maybe we should deduplicate and do the additional lookups instead.